src/Infrastructure.Styles.Analyzer/RMSYN1006LineIsTooLongAnalyzer.cs

```diff
@@ -0,0 +1,85 @@
+﻿// This file is part of the re-motion Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// re-motion Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+
+using System;
+using System.Collections.Immutable;
+using Infrastructure.Styles.Analyzer.Infrastructure;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Infrastructure.Styles.Analyzer
+{
+  [DiagnosticAnalyzer(LanguageNames.CSharp)]
+  public class RMSYN1006LineIsTooLongAnalyzer : DiagnosticAnalyzer
+  {
+    public const string DiagnosticId = "RMSYN1006";
+
+    private static readonly DiagnosticDescriptor Descriptor = new(
+        DiagnosticId,
+        "Line should not be too long",
+        "Line is too long",
+        "Formatting",
+        DiagnosticSeverity.Warning,
+        true);
+
+    /// <inheritdoc />
+    public override void Initialize(AnalysisContext context)
+    {
+      context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+      context.EnableConcurrentExecution();
+
+      context.RegisterSyntaxTreeAction(Analyze);
+    }
+
+    /// <inheritdoc />
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
+        ImmutableArray.Create(Descriptor);
+
+    private static void Analyze(SyntaxTreeAnalysisContext analysisContext)
+    {
+      var maxLineLength = analysisContext.GetAnalyzerSetting(AnalyzerSetting.MaxLineLength);

```

Comment:
Besides, adding another ctor for easier testing would be an idea.

Reply:
Hmm, I need to look into the lifecycle of analyzers for this. We cannot use a single lazy static because the property value is dependent on the file it is in. Different files might have different settings for the max line length.

Reply:
Makes sense, forgot about the inheritance aspect of the editorconfig

Reply:
Yeah, since the `Analyze` method is executed per file we have no choice other than checking the option value